crates/cxx-qt-gen/src/parser/cxxqtdata.rs

```diff
@@ -126,8 +127,11 @@ impl ParsedCxxQtData {
     }
 
     fn parse_foreign_mod_rust_qt(&mut self, mut foreign_mod: ItemForeignMod) -> Result<()> {
-        let namespace = attribute_find_path(&foreign_mod.attrs, &["namespace"])
-            .map(|index| expr_to_string(&foreign_mod.attrs[index].meta.require_name_value()?.value))
+        let attrs = parse_attributes(&foreign_mod.attrs, &["namespace"])?;
+
+        let namespace = attrs
+            .get("namespace")
+            .map(|attr| expr_to_string(&attr.meta.require_name_value()?.value))
             .transpose()?
             .or_else(|| self.namespace.clone());
```

Comment:
why does this not use attribute_get_path like elsewhere?

Reply:
Same further down, what is the difference between parse_attributes and attribute_get_path?

I guess is parse_attributes only allows the given attributes and fails if they aren't there whereas attribute_get_path doesn't mind about unknown attributes?

Maybe instead `parse_attributes` should be called `require_attributes` or `parse_require_attributes` or something with `require` in the name so it's clear it fails if none of the attributes in the list are found?

Then places that use `attribute_get_path` should be checked if they should be using the require method?

Reply:
Yeah, I'll change it to require. It will error if an attribute is found that isn't in the allowed list and also creates the Map of all attributes that it does find, so then it does all the collecting in one place, which did clean things up in one or two places. Some places attribute_get_path are still used are because they would require an absurd list of allowed attrs, and I want to search for an attribute but not error.

Reply:
Done